lib/network/level-net/index.js

```diff
@@ -0,0 +1,289 @@
+'use strict'; // eslint-disable-line
+
+const http = require('http');
+const io = require('socket.io');
+const ioClient = require('socket.io-client');
+const sioStream = require('./sio-stream');
+const async = require('async');
+const assert = require('assert');
+
+const serializeError = require('./utils').serializeError;
+const deserializeError = require('./utils').deserializeError;
+
+const DEFAULT_CALL_TIMEOUT_MS = 5000;
+
+class SocketIOConnection {
+    constructor(url, options = {}) {
+        this.logger = options.logger;
+        this.socket = ioClient(url);
+        this.socketStreams = sioStream(this.socket,
+                                       options.logger,
+                                       options.streamMaxPendingAck,
+                                       options.streamAckTimeoutMs);
+        this.socket.on('error', err => {
+            this.logger.warn('connectivity error to storage daemon',
+                             { error: err });
+            return undefined;
+        });
+    }
+
+    /**
+     * @brief internal RPC implementation w/o timeout
+     *
+     * @param {String} remoteCall name of the remote function to call
+     * @param {Array} args list of arguments to the remote function
+     * @param {function} cb callback called when done
+     * @return {undefined}
+     */
+    _call(remoteCall, args, cb) {
+        if (typeof cb !== 'function') {
+            return cb(new Error(`argument cb=${cb} is not a callback`));
```

Comment:
But this is gonna throw! (just kidding, yes you're perfectly right, thanks for spotting it :))